core/connection.ts

```diff
@@ -333,6 +333,36 @@ export class Connection implements IASTNodeLocationWithBlock {
     this.createShadowBlock(true);
   }
 
+  /**
+   * Reconnects this connection to the input with the given name on the given
+   * block. If there is already a connection connected to that input, that
+   * connection is disconnected.
+   *
+   * @param block The block to connect this connection to.
+   * @param inputName The name of the input to connect this connection to.
+   * @returns True if this connection was able to connect, false otherwise.
+   */
+  reconnect(block: Block, inputName: string): boolean {
+    // No need to reconnect if this connection's block is deleted.
+    if (this.getSourceBlock().isDeadOrDying()) return false;
+
+    const connectionParent = block.getInput(inputName)!.connection;
+    const currentParent = this.targetBlock();
+    if (
+      (!currentParent || currentParent === block) &&
+      connectionParent &&
+      connectionParent.targetConnection !== this
```

Comment:
Can't this be:

```suggestion
      connectionParent?.targetConnection !== this
```

Reply:
We need to evaluate `connectionParent` explicitly so that typescript knows it's defined within the body of the `if`.